boundary/collect.py

```diff
@@ -126,58 +129,66 @@ def collect_fn(self):
             # Ignore alias function for now ??
             if decl.function is None:
                 continue
+
+            relpath = os.path.relpath(decl.location.file)
```

Comment:
In most places where we use `os.path.relpath`, we use it to get the signature, i.e.

`(xxx.name, os.path.relpath(xxx.location.file))`

How about introducing a decl_sig method, this could shorten many lines.

```python
def decl_sig(self, decl):
    return (decl.name, os.path.relpath(decl.location.file))
```

Maybe even a little further,

```python
def decl_sig(self, decl, maybe_external):
    if maybe_external is True and decl.function is None:
        return (decl.name, '?')
    else:
        return (decl.name, os.path.relpath(decl.location.file))
```

Reply:
It's a nice suggestion.